### PR TITLE
[OneShot] Update unit tests

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Update the following unit tests:

```yaml
1. tests/sparseml/pytorch/modifiers/obcq ->tests/sparseml/pytorch/pruning/sparsegpt
2. tests/sparseml/pytorch/modifiers/pruning/wanda
3. tests/sparseml/pytorch/modifiers/quantization/test_pytorch.py
4. tests/sparseml/modifiers/pruning/wanda
5. tests/sparseml/modifiers/quantization/test_base.py
6. tests/sparseml/transformers/finetune/data/test_dataset_loading.py
```
- Updated to use the unittest framework, add `unit` and `requires_torch` markers

### CODE REVIEW

**Review by Satrat — APPROVED** [review]
Looks great, I'm liking the new unittest layout, it does make things a lot clearer

**Comment by dbogunowicz** [comment]
@dsikka I see general refactoring here, moving to unittests from pytest, and using the `requires_torch` util. Is there additional RelEng-related context involved as well? Can we talk about this during our 1v1 today?

**Review by dbogunowicz — APPROVED** [review]

**Review by bfineran — APPROVED** [review]

File: tests/sparseml/modifiers/logarithmic_equalization/test_base.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from sparseml.modifiers.logarithmic_equalization.base import (
+    LogarithmicEqualizationModifier,
+)
+from sparseml.modifiers.smoothquant.base import SmoothQuantModifier
+from tests.sparseml.modifiers.conf import setup_modifier_factory
+from tests.testing_utils import requires_torch
+
+
+@pytest.mark.unit
+@requires_torch
```
Comment (Satrat) [inline tests/sparseml/modifiers/logarithmic_equalization/test_base.py:31]: Lol I realize this one is my fault but none of the base tests should require torch

File: tests/sparseml/modifiers/pruning/sparsegpt/test_base.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+from sparseml.core.factory import ModifierFactory
+from sparseml.core.framework import Framework
+from sparseml.modifiers.obcq.base import SparseGPTModifier
+from tests.sparseml.modifiers.conf import setup_modifier_factory
+from tests.testing_utils import requires_torch
+
+
+@pytest.mark.unit
+@requires_torch
```
Comment (Satrat) [inline tests/sparseml/modifiers/pruning/sparsegpt/test_base.py:28]: Same here, anything under tests/sparseml/modifiers shouldn't need torch, torch is needed by tests/sparseml/pytorch/modifiers

File: tests/sparseml/modifiers/pruning/wanda/test_base.py [hunk 3]
```diff
@@ -13,27 +13,38 @@
...
+
 from sparseml.core.factory import ModifierFactory
 from sparseml.core.framework import Framework
 from sparseml.modifiers.pruning.wanda.base import WandaPruningModifier
 from tests.sparseml.modifiers.conf import setup_modifier_factory
+from tests.testing_utils import requires_torch
+
+
+@pytest.mark.unit
+@requires_torch
```
Comment (Satrat) [inline tests/sparseml/modifiers/pruning/wanda/test_base.py:28]: and here

File: tests/sparseml/modifiers/quantization/test_base.py [hunk 4]
```diff
@@ -12,72 +12,89 @@
...
 from sparseml.core.factory import ModifierFactory
 from sparseml.core.framework import Framework
 from sparseml.modifiers.quantization import QuantizationModifier
 from tests.sparseml.modifiers.conf import setup_modifier_factory
+from tests.testing_utils import requires_torch
 
 
-def test_quantization_registered():
-    setup_modifier_factory()
+@requires_torch
```
Comment (Satrat) [inline tests/sparseml/modifiers/quantization/test_base.py:27]: and here

File: tests/sparseml/modifiers/smoothquant/test_base.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+from sparseml.core.factory import ModifierFactory
+from sparseml.core.framework import Framework
+from sparseml.modifiers.smoothquant.base import SmoothQuantModifier
+from tests.sparseml.modifiers.conf import setup_modifier_factory
+from tests.testing_utils import requires_torch
+
+
+@pytest.mark.unit
+@requires_torch
```
Comment (Satrat) [inline tests/sparseml/modifiers/smoothquant/test_base.py:28]: and here

File: tests/sparseml/pytorch/modifiers/logarithmic_equalization/test_pytorch.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+from sparseml.core.model import ModifiableModel
+from sparseml.modifiers.logarithmic_equalization.pytorch import (
+    LogarithmicEqualizationModifierPyTorch,
+)
+from tests.sparseml.pytorch.helpers import LinearNet
+from tests.testing_utils import requires_torch
+
+
+@pytest.mark.unit
+@requires_torch
```
Comment (Satrat) [inline tests/sparseml/pytorch/modifiers/logarithmic_equalization/test_pytorch.py:31]: and here

**Review by Satrat — CHANGES_REQUESTED** [review]
Should've caught this on the first review, but there are requires_torch in places that it shouldnt be needed. Otherwise this looks good to me!

**Review by Satrat — DISMISSED** [review]

**Review by bfineran — DISMISSED** [review]

**Review by Satrat — APPROVED** [review]

File: .github/workflows/test-check.yaml [hunk 7]
```diff
@@ -165,7 +165,7 @@ jobs:
       - name: "Clean sparsezoo directory"
         run: rm -r sparsezoo/
       - name: "⚙️ Install dependencies"
-        run: pip3 install .[dev,torchvision,onnxruntime]
+        run: pip3 install .[dev,torchvision,onnxruntime] safetensors>=0.4.1
```
[the remark this answers is not in the document]
Comment (dsikka) [reply .github/workflows/test-check.yaml:168]: it's already in transformers. We dont need the other dependencies (at least no until my next PR)

**Review by bfineran — APPROVED** [review]

**Pull request merged**